### ISSUE TITLE
"shadow-cljs - aot compile" taking too long

### ISSUE THREAD

**Issue by thheller** [thheller/shadow-cljs#160]
AOT compilation is now done on install but takes forever on Windows. I based this on my Mac where it takes about 10sec to AOT compile everything but it takes a  lot longer on Windows (1min+). Not sure why it is so slow on Windows.

Maybe should consider publishing the AOT compiled version so the user doesn't have to AOT compile at all. This takes the download size on shadow-cljs.jar from ~300kb-ish to 7mb+ though which isn't great either given the constant new releases I'm doing.

**Comment by jiyinyiyong** [comment]
And downloading npm packages from mainland China would be even slower.

**Comment by thheller** [comment]
I disabled the automatic AOT since it was causing issues with heroku as well.

Thinking about other dist options ...

-  standalone "uberjar" + AOT would be about 34MB.
- AOT jar 8MB
- "just sources" 377KB

I would like to stick to just distributing the sources but AOT makes a really huge difference in startup time which has a huge impact on user experience. So AOT is sort of required. Question is whether it is better to wait for a 8MB download or wait for AOT compilation. In most cases the 8MB download should be a whole lot faster. So think it is the better option.

The user currently also has to download the separate 10MB `shadow-cljs-jar`. `tools.deps` does a very similar thing but is also 19MB. Either way this separate download is required unless publishing as an "uberjar".

"uberjar" is not a great option though since nothing from `maven` is shared this way and you are downloading everything with each new version. Currently a new version usually just means downloading the small jar unless I changed some deps. A lot of other headaches go away though since it is much easier to isolate dependencies.

Could maybe publish the uberjar to github or some CDN and do a similar model that `lein` does. Small script which downloads the "runtime" on-demand.
